notify/util.go

```diff
@@ -167,6 +167,20 @@ func readAll(r io.Reader) string {
 	return string(bs)
 }
 
+// getFailureStatusCodeCategory return the status code category for failure request
+// the status starts with 4 will return 4xx and starts with 5 will return 5xx
+// other than 4xx and 5xx input status will return an error.
+func getFailureStatusCodeCategory(statusCode int) (string, error) {
+	if statusCode/100 == 4 {
+		return "4xx", nil
+	}
+	if statusCode/100 == 5 {
+		return "5xx", nil
+	}
+
+	return "", fmt.Errorf("unexpected status code %v", statusCode)
```

Comment:
Do we really want to fail some application logic if status code is unknown? I would just return `unknown` string here.